.github/actions/nexus-count-artifacts/action.yml

```diff
@@ -0,0 +1,50 @@
+name: Count artifacts in Nexus Repository
+description: Count artifacts in Nexus Repository
+inputs:
+  nexus-username:
+    description: Nexus username
+    required: true
+  nexus-password:
+    description: Nexus password
+    required: true
+  nexus-url:
+    description: Base URL to the Nexus server
+    required: true
+  repository:
+    description: The repository to count artifacts
+    required: true
+  group:
+    description: The maven group-id of the components to be moved
+    required: true
+  version:
+    description: The version of the components to be moved
+    required: true
+outputs:
+  artifact_count:
+    description: The number of artifacts in the repository
+    value: ${{ steps.count-artifacts.outputs.artifact_count }}
+
+runs:
+  using: composite
+  steps:
+    - name: count artifacts in Nexus
+      id: count-artifacts
+      shell: bash
+      env:
+        NEXUS_USERNAME: ${{ inputs.nexus-username }}
+        NEXUS_PASSWORD: ${{ inputs.nexus-password }}
+        NEXUS_URL: ${{ inputs.nexus-url }}
+        REPOSITORY: ${{ inputs.repository }}
+        GROUP: ${{ inputs.group }}
+        VERSION: ${{ inputs.version }}
+
+      run: |
+        echo "Counting artifacts in repository ${REPOSITORY} in Nexus"
+        search_response=$(curl -sSf -u "${NEXUS_USERNAME}:${NEXUS_PASSWORD}" \
+          -X GET "${NEXUS_URL}/service/rest/v1/search?repository=${REPOSITORY}&group=${GROUP}&version=${VERSION}" \
+          -H 'Content-Type: application/json' \
+          -H 'accept: application/json' \
+          -d '{}')
+        artifact_count=$(echo "$search_response" | yq '.items | length')
+        echo "Artifact Count: $artifact_count"
```

Comment:
Actually, depending on the number of the artifacts the results can be [paged](https://help.sonatype.com/en/pagination.html). This is not problem for our use case where we just need to know if there are elements to move or not, but I was wondering if we should add a marker in case where the `continuationToken` is not `null` (meaning that there are more elements than the ones returned). For instance, return `100` if  `continuationToken` is `null` and return `100+` if it's not `null`. 

Alternatively, we could embed the call directly in the action `nexus-move-artifact` instead of having a dedicated one for counting.

What do you think @revatilimaye @atchertchian @gionn ?

Reply:
I have a preference for this action being embedded in the "move" one

Reply:
Have embedded the count-artifacts action in move-artifacts